src/store/Chat/index.js

```diff
@@ -161,9 +180,34 @@ function ChatReducer (state = initialState, action) {
       //   });
       // }
 
+      // We need to run through the entire array of messages and aggregate them
+      //  into similar sets
+
+      if (action.payload.length > 0) {
+        sortedPayload = action.payload.sort((curr, next) => (
+          new Date(curr.timestamp) - new Date(next.timestamp)
+        ));
+        // There should be an algorithm here that would speed things up
```

Comment:
I meant the former. Your reasoning makes total sense. But let's pretend I also meant the latter so that I sound smart too. 

👍